packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -1,23 +1,40 @@
 import cxapi = require('@aws-cdk/cx-api');
+import { makeUniqueId } from '../util/uniqueid';
 export const PATH_SEP = '/';
 
 /**
  * Represents the building block of the construct graph.
  * When a construct is created, it is always added as a child
  */
 export class Construct {
-    private static readonly VALID_NAME_REGEX = /^[A-Za-z][A-Za-z0-9]*$/;
-
     /**
      * Returns the parent of this node or undefined if this is a root node.
      */
     public readonly parent?: Construct;
 
     /**
-     * The name of this construct
+     * @deprecated use `id`
      */
     public readonly name: string;
 
+    /**
+     * The subtree-local id of the construct.
+     * This id is unique within the subtree. To obtain a tree-unique id, use `uniqueId`.
```

Comment:
Define `subtree`... The construct may be *accessible* from several parts of the tree, but I suppose the `subtree` in question is the one it's attached to, not the one you're accessing from...

Reply:
Changed description to:

```ts
    /**
     * The local id of the construct.
     * This id is unique amongst its siblings.
     * To obtain a tree-global unique id for this construct, use `uniqueId`.
     */
    public readonly id: string;
```